pkg/operation/botanist/kubernetesdashboard.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/kubernetesdashboard"
+	"github.com/gardener/gardener/pkg/utils/images"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	"k8s.io/utils/pointer"
+)
+
+// DefaultKubernetesDashboard returns a deployer for kubernetes-dashboard.
+func (b *Botanist) DefaultKubernetesDashboard() (component.DeployWaiter, error) {
+	image, err := b.ImageVector.FindImage(images.ImageNameKubernetesDashboard, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	scraperImage, err := b.ImageVector.FindImage(images.ImageNameKubernetesDashboardMetricsScraper, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	values := kubernetesdashboard.Values{
+		Image:               image.String(),
+		MetricsScraperImage: scraperImage.String(),
+		VPAEnabled:          b.Shoot.WantsVerticalPodAutoscaler,
+		KubernetesVersion:   b.Shoot.KubernetesVersion,
+	}
+
+	if b.APIServerSNIEnabled() {
+		values.APIServerHost = pointer.String(b.outOfClusterAPIServerFQDN())
+	}
+
+	if b.Shoot.GetInfo().Spec.Addons.KubernetesDashboard.AuthenticationMode != nil {
+		values.AuthenticationMode = *b.Shoot.GetInfo().Spec.Addons.KubernetesDashboard.AuthenticationMode
+	}
+
+	return kubernetesdashboard.New(b.SeedClientSet.Client(), b.Shoot.SeedNamespace, values), nil
+}
+
+// DeployKubernetesDashboard deploys the Kubernetes Dashboard component.
+func (b *Botanist) DeployKubernetesDashboard(ctx context.Context) error {
```

Comment:
Is it possible to add unit tests for this func? The unit tests should be similar to https://github.com/gardener/gardener/blob/b4df8668a47e3c68be00a55a9692b60242ab7fbb/pkg/operation/botanist/vpa_test.go#L117-L166.

Reply:
For this, mock client will be required. I think it is not important so we can skip it.

Reply:
We will need to generate a mock for the kubernetesDashboard component similar to https://github.com/gardener/gardener/blob/05a47f7edbf59448be9f5ad69dd00a547a4832d4/pkg/operation/botanist/component/vpa/mock/doc.go#L15-L17.
We can still add the unit test as for the VPA case - the VPA has similar `DeployVerticalPodAutoscaler ` func that depending on whether the VPA is enabled deploys/destroys the component. And this func is unit tested.

Reply:
Yes, But this doesn't add much value that's why I don't want to add it.

Reply:
There is value and it is that `Destroy` will be called instead of `Deploy` when the addon is not enabled in the Shoot spec. The fact that the `DeployKubernetesDashboard` func is simple does not mean that it does not have to be unit tested.